service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/ObservationMapper.java

```diff
@@ -111,8 +114,16 @@ private String prepareCodeElement(Observation observation) {
         return StringUtils.EMPTY;
     }
 
+    private ClassCode prepareClassCode(Observation observation) {
+        if (observation.hasRelated()) {
+            return ClassCode.BATTERY;
+        }
+
+        return ClassCode.CLUSTER;
+    }
+
     private String prepareObservationStatement(IdMapper idMapper, Observation observation) {
-        if (observation.hasCode()) {
+        if (!CodeableConceptMappingUtils.hasCode(observation.getCode(), List.of(NARRATIVE_STATEMENT_CODE))) {
```

Comment:
This looks like `CodeableConceptMappingUtils#hasCode` could use an overload accepting varargs. What do you think?

Reply:
A number of places would need updated to make this change. Also since it'll then be treated as an array, we'd need to update the implementation of `CodeableConceptMappingUtils.hasCode` to check where the String equals "any String in array". I don't think this is worth the effort as part of this ticket